cloudinit/net/dhcp.py

```diff
@@ -267,101 +132,366 @@ def networkd_get_option_from_leases(keyname, leases_d=None):
+                current_idx = idx + req_toks
+            elif net_length in range(9, 17):
+                req_toks = 7
+                if len(tokens[idx:]) < req_toks:
+                    _trunc_error(net_length, req_toks, len(tokens[idx:]))
+                    return static_routes
+                net_address = ".".join(tokens[idx + 1 : idx + 3] + ["0", "0"])
+                gateway = ".".join(tokens[idx + 3 : idx + req_toks])
+                current_idx = idx + req_toks
+            elif net_length in range(1, 9):
+                req_toks = 6
+                if len(tokens[idx:]) < req_toks:
+                    _trunc_error(net_length, req_toks, len(tokens[idx:]))
+                    return static_routes
+                net_address = ".".join(
+                    tokens[idx + 1 : idx + 2] + ["0", "0", "0"]
+                )
+                gateway = ".".join(tokens[idx + 2 : idx + req_toks])
+                current_idx = idx + req_toks
+            elif net_length == 0:
+                req_toks = 5
+                if len(tokens[idx:]) < req_toks:
+                    _trunc_error(net_length, req_toks, len(tokens[idx:]))
+                    return static_routes
+                net_address = "0.0.0.0"
+                gateway = ".".join(tokens[idx + 1 : idx + req_toks])
+                current_idx = idx + req_toks
+            else:
+                LOG.error(
+                    'Parsed invalid net length "%s".  Verify DHCP '
+                    "rfc3442-classless-static-routes value.",
+                    net_length,
+                )
+                return static_routes
 
-    return static_routes
+            static_routes.append(
+                ("%s/%s" % (net_address, net_length), gateway)
+            )
 
+        return static_routes
+
+    @staticmethod
+    def get_dhclient_d():
+        # find lease files directory
+        supported_dirs = [
+            "/var/lib/dhclient",
+            "/var/lib/dhcp",
+            "/var/lib/NetworkManager",
+        ]
+        for d in supported_dirs:
+            if os.path.exists(d) and len(os.listdir(d)) > 0:
+                LOG.debug("Using %s lease directory", d)
+                return d
+        return None
+
+    @staticmethod
+    def get_latest_lease(lease_d=None):
+        # find latest lease file
+        if lease_d is None:
+            lease_d = IscDhclient.get_dhclient_d()
+        if not lease_d:
+            return None
+        lease_files = os.listdir(lease_d)
+        latest_mtime = -1
+        latest_file = None
+
+        # lease files are named inconsistently across distros.
+        # We assume that 'dhclient6' indicates ipv6 and ignore it.
+        # ubuntu:
+        #   dhclient.<iface>.leases, dhclient.leases, dhclient6.leases
+        # centos6:
+        #   dhclient-<iface>.leases, dhclient6.leases
+        # centos7: ('--' is not a typo)
+        #   dhclient--<iface>.lease, dhclient6.leases
+        for fname in lease_files:
```

Comment:
It could be, but I'm not going to prioritize it right now. Right now my focus is getting functionality correct without risking breakage of things that currently work.